src/datatype/svecf32.rs

```diff
@@ -0,0 +1,698 @@
+    let mut values = Vec::<F32>::new();
+    let mut state = MatchingLeft;
+    let mut token: Option<String> = None;
+    let mut index = 0;
+    for &c in input {
+        match (state, c) {
+            (MatchingLeft, b'[') => {
+                state = Reading;
+            }
+            (Reading, b'0'..=b'9' | b'a'..=b'z' | b'A'..=b'Z' | b'.' | b'+' | b'-') => {
+                let token = token.get_or_insert(String::new());
+                token.push(char::from_u32(c as u32).unwrap());
+            }
+            (Reading, b',') => {
+                let token = solve(token.take(), "Expect a number.");
+                let value: F32 = solve(token.parse().ok(), "Bad number.");
+                if !value.is_zero() {
+                    indexes.push(index);
+                    values.push(value);
+                }
+                index = match index.checked_add(1) {
+                    Some(x) => x,
+                    None => SessionError::BadValueDimensions.friendly(),
+                };
+            }
+            (Reading, b']') => {
+                if let Some(token) = token.take() {
+                    let value: F32 = solve(token.parse().ok(), "Bad number.");
+                    if !value.is_zero() {
+                        indexes.push(index);
+                        values.push(value);
+                    }
+                    index = match index.checked_add(1) {
+                        Some(x) => x,
+                        None => SessionError::BadValueDimensions.friendly(),
+                    };
+                }
+                state = MatchedRight;
+            }
+            (_, b' ') => {}
+            _ => {
+                SessionError::BadLiteral {
+                    hint: format!("Bad character with ascii {:#x}.", c),
+                }
+                .friendly();
+            }
+        }
+    }
+    if state != MatchedRight {
+        SessionError::BadLiteral {
+            hint: "Bad sequence.".to_string(),
+        }
+        .friendly();
+    }
+    SVecf32::new_in_postgres(SparseF32Ref {
+        dims: index,
+        indexes: &indexes,
+        values: &values,
+    })
+}
+
+#[pgrx::pg_extern(immutable, parallel_safe, strict)]
+fn _vectors_svecf32_out(vector: SVecf32Input<'_>) -> CString {
+    let mut buffer = String::new();
+    buffer.push('[');
+    let vec = vector.data().to_dense();
+    let mut iter = vec.iter();
+    if let Some(x) = iter.next() {
+        buffer.push_str(format!("{}", x).as_str());
+    }
+    for x in iter {
+        buffer.push_str(format!(", {}", x).as_str());
+    }
+    buffer.push(']');
+    CString::new(buffer).unwrap()
+}
+
+#[cfg(any(feature = "pg14", feature = "pg15", feature = "pg16"))]
+#[pgrx::pg_extern(sql = "\
+CREATE FUNCTION _vectors_svecf32_subscript(internal) RETURNS internal
+IMMUTABLE STRICT PARALLEL SAFE LANGUAGE c AS 'MODULE_PATHNAME', '@FUNCTION_NAME@';")]
+fn _vectors_svecf32_subscript(_fcinfo: pgrx::pg_sys::FunctionCallInfo) -> Datum {
```

Comment:
Copy the test file.

Reply:
Previously, there is `tests/sqllogictest/pg14/svector_subscript.slt`.